public/casestudies/adeo.yml

```diff
@@ -0,0 +1,92 @@
+
+    Initial solution was a developer portal with list of all applications and reference to dedicated documentation. Some legacy systems had docs written in MS Excel.
+
+    There was a need for a standart way of describing event-driven architecture.
+solution: |
+    The API is now described with AsyncAPI. AsyncAPI file, stored together with the source code, is used to generate HTML documentation in the same release pipeline for the product. Documentation is exposed as part of the product internally for different units and external partners.
+
+    Payloads are described with Avro schema. These schemas are used to generate models and they are also referenced directly in AsyncAPI files thanks to `schemaFormat` feature and `$ref`. This way they make sure code is aligned with docs.
+
+    Shift to using AsyncAPI also enables the team to implement more use cases using AsyncAPI files.
+technical:
+  language:
+    - Java
+  frameworks:
+    - Spring
+  protocol:
+    - Kafka
+  broker: TODO: size of cluster
+  testing: TODO: how is it done
+  architecture: |
+    The following [enterprise integration patterns](https://www.enterpriseintegrationpatterns.com/patterns/messaging) are applied:
+    - [Request/reply](https://www.enterpriseintegrationpatterns.com/patterns/messaging/RequestReply.html)
+      Described with `description` field in AsyncAPI. Reply goes to dedicated reply channel.
+      TODO: Can we have a snippet?  
+    - [Return address](https://www.enterpriseintegrationpatterns.com/patterns/messaging/ReturnAddress.html)
+      Message Header object with `REPLY_TOPIC` header property. Info that needs to be provided by client, so producer knows where to send a response.
+      TODO: Can we have a snippet? 
+    - [Correlation identifier](https://www.enterpriseintegrationpatterns.com/patterns/messaging/CorrelationIdentifier.html)
+      Important to follow to be able to identify what request does the respons match.
+      `REQUEST_ID` in Message Header object, and pointed out with `correlationID` property. 
+      TODO: I could not figure out what role `CORRELATION_ID` is playing. Is it that once producer sends a message to reply channel, then info about request id is not in `REQUEST_ID` but `CORRELATION_ID`. So they both have the same info, but one is for request producer and the other for reply producer?
+      TODO: Can we have a snippet? 
+    - [DeadLetter channel](https://www.enterpriseintegrationpatterns.com/patterns/messaging/DeadLetterChannel.html)
+      Also known as Dead Letter Queue. In Kafka it is just another channel where undelivered messages are sent. In AsyncAPI for now it is just part of channel description, channel that works as DeadLetter channel.
+      TODO: Can we have a snippet? 
+codegen: |
+  Java models generation. Avro schemas used as a source.
```

Comment:
I wonder if we can get this info as an object. Something like:

```yaml
codegen:
  - language: java
    target: models # Can also be "projects". Something else?
    source: avro
```

Reply:
hehe, to be honest, this was my first version before creating the PR 😄 
but then I decided to make it more generic because it is a first case study, and we can always change from description to more structured object later because:
- this is not like the AsyncAPI spec, where we do not own user's documents. We own these case studies. So we can improve the template, and provided case studies with every new one, as we want
- for the same reason ☝🏼 better to improve the template and content on the fly, than now having a discussion if `target` should be enum of `models` and `projects`, or maybe `models` and `clients` and `application`. 


Thoughts?